server/lib/JJ.pm

```diff
@@ -0,0 +1,31 @@
+package JJ;
```

Comment:
Thank you for reviewing this PR.

Obviously the logging code was not meant to be pulled "as is". So please ignore it.
I nonetheless think a logging facility is of great help when debugging an lsp server and will leave it to you to implement it if you feel its convenient to your project.
Some lsp servers use a command line "--log-file" flag for this purpose or maybe some "workspace/configuration" could be used too.

Oh, and JJ is an acronym for "stupid me", a useful trick to mark code I messed with because "JJ" is seldom found in code bases. (Incidentally those are my initials).